core/src/main/java/org/radargun/stages/stream/StreamStage.java

```diff
@@ -0,0 +1,214 @@
+
+   @Property(doc = "The number of times to execute the stream task. The default is 10.")
+   private int numExecutions = 10;
+
+   @Property(doc = "Name of the cache where stream task should be executed. Default is the default cache.")
+   private String cacheName;
+
+   @Property(doc = "The name of the key in the MasterState object that returns the total number of "
+           + "bytes processed by the stream task. The default is RandomDataStage.RANDOMDATA_TOTALBYTES_KEY.")
+   private String totalBytesKey = RandomDataStage.RANDOMDATA_TOTALBYTES_KEY;
+
+   @Property(optional = false, doc = "Fully qualified class name of the StreamConsumer implementation.")
+   private String streamOperationClass;
+
+   @InjectTrait
+   private Streamable streamable;
+
+   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   private Clustered clustered;
+
+   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   private CacheInformation cacheInformation;
+
+   @InjectTrait(dependency = InjectTrait.Dependency.MANDATORY)
+   private BasicOperations basicOperations;
+
+   private Streamable.StreamFunction function;
+
+   private Object streamTaskResult;
+
+   @Override
+   public StageResult processAckOnMaster(List<DistStageAck> acks) {
+      StageResult result = super.processAckOnMaster(acks);
+      if (result.isError())
+         return result;
+
+      Report report = masterState.getReport();
+      Report.Test test = report.createTest("Stream_Stage", null, true);
+      int testIteration = test.getIterations().size();
+
+      Map<Integer, Report.SlaveResult> numberOfResultKeysResult = new HashMap<>();
+      Map<Integer, Report.SlaveResult> durationsResult = new HashMap<>();
+
+      for (StreamStageAck ack : Projections.instancesOf(acks, StreamStageAck.class)) {
+         if (ack.stats != null) {
+            DataOperationStats opStats = (DataOperationStats) ack.stats.getOperationsStats().get(Streamable.STREAMABLE.name);
+
+            opStats.setTotalBytes((Long) masterState.get(totalBytesKey));
+            test.addStatistics(testIteration, ack.getSlaveIndex(), Collections.singletonList(ack.stats));
+            durationsResult.put(ack.getSlaveIndex(), new Report.SlaveResult(opStats.getResponseTimes(), false));
+            test.addResult(testIteration, new Report.TestResult("Stream task durations", durationsResult, "", false));
+         }
+         if (ack.numberOfResultKeys != null) {
+            numberOfResultKeysResult.put(ack.getSlaveIndex(), new Report.SlaveResult(ack.numberOfResultKeys, false));
+            test.addResult(testIteration, new Report.TestResult("Key count in stream result map",
+                    numberOfResultKeysResult, "", false));
+         }
+      }
+      return StageResult.SUCCESS;
+   }
+
+   @Override
+   public DistStageAck executeOnSlave() {
+      if (!isServiceRunning()) {
+         return errorResponse("Service is not runnning", null);
+      }
+
+      if (slaveState.getSlaveIndex() == 0) {
```

Comment:
@alanfx @rvansa @rmacor Is there any reason to make the slave index hardcoded? This is what I've recently observed in MapReduce stage as well. If it's supposed to run only on one slave, it should be documented in stage documentation and configured by "slaves" attribute.